### PR TITLE
 BaSP-TG-3:Delete-and-update-employee

### PR DESCRIPTION
**Pull request by GonzaloAsencio**
- Add a new model for employees.
- Add/Update the router implementation to use the correct router based on the entity.
- Add route /employees/:id to delete an specific employee by id .
- Add route /employees/:id to update a new employee and update the controller to make the changes in the database.
- Add validations for the employee update.
- Update the postman collection.

### CODE REVIEW

File: src/controllers/employees.js [hunk 1]
```diff
@@ -1,96 +1,51 @@
...
-    if (filterByName) {
-      filterdList = employees.filter((user) => user.first_name.includes(filterByName));
-    }
-    if (filterByDni) {
-      filterdList = employees.filter((user) => user.dni.toString().includes(filterByDni));
+export const deleteEmployee = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const result = await Employees.findByIdAndDelete(id);
+    if (!result) {
```
Comment (lmalvasia) [inline src/controllers/employees.js:7]: This validation is not necessary in this case. When the employee does not exist, the `findByIdAndDelete` method returns an error, so we are going to return the error that we have inside the `catch` block.

File: src/controllers/employees.js [hunk 2]
```diff
@@ -1,96 +1,51 @@
...
-  fs.writeFile('src/data/employees.json', JSON.stringify(deleteUser), (err) => {
-    if (err) {
-      res.send('Cannot deleted user');
-    } else {
-      res.status(200).json({
-        message: 'Employee deleted',
+export const editEmployee = async (req, res) => {
+  try {
+    const { id } = req.params;
+    const result = await Employees.findByIdAndUpdate(
```
Comment (lmalvasia) [inline src/controllers/employees.js:29]: Use this:
await Employees.findByIdAndUpdate(id, req.body, { new: true });

Comment (lmalvasia) [reply src/controllers/employees.js:29]: It's not necessary to send the `_id` with an object and to use a destructuring from req.body.

File: src/validations/employees.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    return res.status(400).json({
+      message: `There was an error:${validation.error.details[0].message}`,
+      data: undefined,
+      error: true,
+    });
+  }
+  return next();
+};
+
+export default {
```
Comment (lmalvasia) [inline src/validations/employees.js:23]:
```suggestion
export default validateEmployeesBody;
```

File: src/controllers/employees.js (hunk 1 above)
Comment (GonzaloAsencio) [reply src/controllers/employees.js]: But if you try to validate a wrong id-employee, the method return " message: "Employee with ID deleted" 
anyway. Fixed!

File: src/controllers/employees.js (hunk 2 above)
Comment (GonzaloAsencio) [reply src/controllers/employees.js]: Oh i see... Fixed!

File: src/validations/employees.js (hunk 3 above)
Comment (GonzaloAsencio) [reply src/validations/employees.js]: Fixed

**Review by GinaSchP — APPROVED** [review]

**Pull request merged**